src/sgame/sg_bot_util.cpp

```diff
@@ -2312,7 +2312,11 @@ botTarget_t& botTarget_t::operator=(const gentity_t *newTarget) {
 
 	if (!targetsValidEntity())
 	{
-		Log::Warn("bot: selecting invalid entity as target");
+		Log::Warn("bot: selecting invalid entity as target, %s",
+			!newTarget->inuse ? "entity isn't allocated" :
+			!Entities::IsAlive(newTarget) ? "entity is dead" :
+				"for some reason" );
+		ASSERT_UNREACHABLE(); // this will be a bug in the calling code, let's warn about it
```

Comment:
This actually can make the program crash in release mode, probably not what you want

Reply:
Is ASSERT(0) different in crash behaviour?

Reply:
Looks like it is actually different with GCC: ASSERT(0) preprocesses to nothing while ASSERT_UNREACHABLE makes the behavior undefined there (the compiler doesn't ensure that it will jump to the logically next part of the code). 

The main thing is it doesn't make sense to have both a warning and an assert. You should pick one or the other.

Reply:
Well the warning and the assert have different purpose: the warning makes it explicit what failed, and the assert makes it possible to have a backtrace.

Reply:
I've changed it to `ASSERT(0);`

Reply:
Assertions should only be used for a fatally bad bug like indexing out of bounds. Making it an assert forces the developer to do something about this bot bug even if he is trying to work on something else. If you get a warning and want to debug it, you can put a breakpoint on the log statement for the next time it happens.

Reply:
Ok, fair enough. I've removed the assertion.